### PR TITLE
Update LXD URL

### PR DESCRIPTION
**Pull request by s-makin**
```
LXD documentation has moved and broken this link as a result.
I believe this is the correct one to redirect to, but please
advise if not.
```

## Checklist:

 - [ ] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/contributing.html)
 - [ ] I have updated or added any unit tests accordingly
 - [X] I have updated or added any documentation accordingly

### CODE REVIEW

File: cloudinit/config/cc_lxd.py [hunk 1]
```diff
@@ -79,7 +79,7 @@
             # storage_pools, profiles, projects, clusters and core config,
             # `lxd:preseed` config will be passed as stdin to the command:
             #  lxd init --preseed
-            # See https://linuxcontainers.org/lxd/docs/master/preseed/ or
+            # See https://documentation.ubuntu.com/lxd/en/latest/howto/initialize/#non-interactive-configuration or
```
Comment (aciba90) [inline cloudinit/config/cc_lxd.py:82]: I think might fix it:

```suggestion
            # See 
            # https://documentation.ubuntu.com/lxd/en/latest/howto/initialize/#non-interactive-configuration
            # or
```

**Review by aciba90 — CHANGES_REQUESTED** [review]
LGTM but CI complains: https://github.com/canonical/cloud-init/actions/runs/5462632134/jobs/9942252697?pr=4223

File: cloudinit/config/cc_lxd.py (hunk 1 above)
Comment (aciba90) [reply cloudinit/config/cc_lxd.py:82]: or:

```suggestion
            # See https://documentation.ubuntu.com/lxd/en/latest/howto/initialize/#non-interactive-configuration or  # noqa
```

Comment (s-makin) [reply cloudinit/config/cc_lxd.py:82]: I'm not sure I understand the difference between the two - which one is preferred?

Comment (aciba90) [reply cloudinit/config/cc_lxd.py:82]: Sorry for the lack of explanation. They were clumsy ways to disable the [line too long](https://github.com/canonical/cloud-init/actions/runs/5462632134/jobs/9942252697?pr=42230) complaint from flake8.

The problem is that those lines get rendered in the final docs, so anything we append to the line will get rendered in the docs.

As a probably better way to disable the lint error, I would suggest to apply the following diff:

```diff
diff --git a/cloudinit/config/cc_lxd.py b/cloudinit/config/cc_lxd.py
index 2b8837aac..5dcb620b1 100644
--- a/cloudinit/config/cc_lxd.py
+++ b/cloudinit/config/cc_lxd.py
@@ -149,7 +149,7 @@ meta: MetaSchema = {
                     name: limited
 
 
-            """
+            """  # noqa: E501
         ),
     ],
     "frequency": PER_INSTANCE,

```

This disables the E501 error on the whole string block without affecting the rendered docs.

Comment (s-makin) [reply cloudinit/config/cc_lxd.py:82]: Thanks for the suggestion - and the explanation! :) I've changed that line as per your comment

**Review by aciba90 — APPROVED** [review]
+1 thanks!

**Comment by aciba90** [comment]
@s-makin, something weird is going on here, the new link is not in [reference/modules.html#lxd](https://canonical-cloud-init.readthedocs-hosted.com/en/latest/reference/modules.html#lxd), or am I missing something?

**Pull request merged**